pkg/modules/role/role.go

```diff
@@ -0,0 +1,66 @@
+package role
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/SigNoz/signoz/pkg/types/authtypes"
+	"github.com/SigNoz/signoz/pkg/types/roletypes"
+	"github.com/SigNoz/signoz/pkg/valuer"
+)
+
+type Module interface {
+	// Creates the role metadata
+	Create(context.Context, valuer.UUID, *roletypes.PostableRole) (*roletypes.GettableRole, error)
```

Comment:
```suggestion
	// Creates the role metadata
	Create(context.Context, valuer.UUID, <split-this>) (*roletypes.Role, error)
```
Postable, Gettable and Updatable types are just for the http layer. The Module works on a central entity `Role`. It can take a role and return a role. It can internally play with StorableRole.

Flow
```
PostableRole -> decomposed -> passed to module -> storable role -> role -> sent back to handler -> gettable role -> user
```

Everyone needs to have Role. Now if Role == StorableRole, that is okay. If Role == GettableRole, that is also okay.